frontend/src/components/Room.tsx

```diff
@@ -73,6 +78,31 @@ class Room extends React.Component<Props, State> {
     this.setUsernameAndEmit = this.setUsernameAndEmit.bind(this);
     this.setModalState = this.setModalState.bind(this);
     this.changeUsernameAndEmit = this.changeUsernameAndEmit.bind(this);
+    this.handleSetUsers = this.handleSetUsers.bind(this);
+    this.handleSetUsername = this.handleSetUsername.bind(this);
+  }
+
+  //Creates an object of users { key: id, value : username}
```

Comment:
creating a type interface or use a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead

Reply:
Not sure what you mean here... I am creating a map of users from the given array?

Reply:
I mean using a Map in JavaScript https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map rather than an object or an TypeScript interface with indexed types https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map